governance/program/src/processor/process_create_mint_governance.rs

```diff
@@ -84,13 +88,28 @@ pub fn process_create_mint_governance(
         rent,
     )?;
 
-    if transfer_mint_authority {
-        set_spl_token_mint_authority(
+    if transfer_mint_authorities {
+        set_spl_token_account_authority(
             governed_mint_info,
             governed_mint_authority_info,
             mint_governance_info.key,
+            AuthorityType::MintTokens,
             spl_token_info,
         )?;
+
+        // If the mint has freeze_authority then transfer it as well
+        let mint_data = Mint::unpack(&governed_mint_info.data.borrow())?;
+        // Note: The code assumes mint_authority==freeze_authority
+        //       If this is not the case then the caller should set freeze_authority accordingly before making the transfer
+        if mint_data.freeze_authority.is_some() {
```

Comment:
With a separate flag, you can check that the freeze authority is the given `governed_mint_authority_info`, and give a clear error if it isn't

Reply:
Agree that would be better, but since the instruction is going to be deprecated anyway I just did the minimum to ensure the partial transfer of authorities is not possible.

Reply:
I don't think this is the correct choice, but if it's just a stop-gap, then no problem